pathology/hovernet/infer_roi.py

```diff
@@ -0,0 +1,217 @@
+import logging
+import os
+import time
+from argparse import ArgumentParser
+from glob import glob
+
+import torch
+import torch.distributed as dist
+
+from monai.apps.pathology.inferers import SlidingWindowHoVerNetInferer
+from monai.apps.pathology.transforms import (
+    GenerateDistanceMapd,
+    GenerateInstanceBorderd,
+    GenerateWatershedMarkersd,
+    GenerateWatershedMaskd,
+    HoVerNetNuclearTypePostProcessingd,
+    Watershedd,
+)
+from monai.data import DataLoader, Dataset, PILReader
+from monai.engines import SupervisedEvaluator
+from monai.networks.nets import HoVerNet
+from monai.transforms import (
+    Activationsd,
+    AsDiscreted,
+    CastToTyped,
+    Compose,
+    EnsureChannelFirstd,
+    FillHoles,
+    FromMetaTensord,
+    GaussianSmooth,
+    LoadImaged,
+    PromoteChildItemsd,
+    SaveImaged,
+    ScaleIntensityRanged,
+)
+from monai.utils import HoVerNetBranch, first
+
+
+def create_output_dir(cfg):
+    if cfg["mode"].lower() == "original":
+        cfg["patch_size"] = 270
+        cfg["out_size"] = 80
+    elif cfg["mode"].lower() == "fast":
+        cfg["patch_size"] = 256
+        cfg["out_size"] = 164
+
+    timestamp = time.strftime("%y%m%d-%H%M%S")
+    run_folder_name = f"{timestamp}_inference_hovernet_ps{cfg['patch_size']}"
+    log_dir = os.path.join(cfg["output"], run_folder_name)
+    print(f"Logs and outputs are saved at '{log_dir}'.")
+    if not os.path.exists(log_dir):
+        os.makedirs(log_dir)
+    return log_dir
+
+
+def run(cfg):
+    # --------------------------------------------------------------------------
+    # Set Directory and Device
+    # --------------------------------------------------------------------------
+    output_dir = create_output_dir(cfg)
+    multi_gpu = True if cfg["use_gpu"] and torch.cuda.device_count() > 1 else False
+    if multi_gpu:
```

Comment:
In the multi-gpu mode, I think here every GPU runs all the same data?
You should (1) partition dataset to every GPU or (2) use `DistributedSampler` in the DataLoader. Right?

Thanks.

Reply:
Oh I know, thanks for catching that. I forgot to implement it.

Reply:
Done!